src/sas/qtgui/Calculators/DataOperationUtilityPanel.py

```diff
@@ -51,6 +54,7 @@ def __init__(self, parent=None):
         # push buttons
         self.cmdClose.clicked.connect(self.onClose)
         self.cmdHelp.clicked.connect(self.onHelp)
+        self.cmdSave.clicked.connect(self.onSave)
```

Comment:
Please add checks for empty data/charts in `onClose`.
Currently clicking on `Save` with empty data gives
```
ERROR: Traceback (most recent call last): File "C:\projects\sasview\src\sas\qtgui\Calculators\DataOperationUtilityPanel.py", line 172, in onSave self.onPrepareOutputData() File "C:\projects\sasview\src\sas\qtgui\Calculators\DataOperationUtilityPanel.py", line 184, in onPrepareOutputData self.output.name = name ^^^^^^^^^^^^^^^^ AttributeError: 'NoneType' object has no attribute 'name'
```

Reply:
Checks for empty data and graphs were added throughout. I also noticed while implementing these that it was possible to save an output from a previous operation before you hit compute. I've added code that resets the output data to None if any of the operation conditions change so that unknown data isn't accidentally sent to the data explorer.